### PR TITLE
Add class description JavaDoc to all public classes

### PR DESCRIPTION
**Pull request by aelred**
Also enforce this with a checkstyle rule

### CODE REVIEW

**Comment by aelred** [comment]
@duckofyork @kasper-piskorski when you have time, can you pull down this branch and add JavaDocs to your public classes? 

Run `mvn compile` to see what needs to be added. You only need it for the class definitions, not for the methods.

**Comment by aelred** [comment]
@kasper-piskorski can you have a look at this when you've got time. There's only three more classes left to comment in `grakn-graql`.

**Comment by fppt** [comment]
@aelred whats the status on this PR?

**Comment by aelred** [comment]
@fppt it's an ongoing effort. Thanks to @duckofyork  and @kasper-piskorski we nearly have JavaDoc for all public graql classes :D

**Comment by aelred** [comment]
@alexandraorth when you have time can you have a look at this PR? We need to add JavaDoc to all public classes.

**Comment by alexandraorth** [comment]
@aelred going to do this now

**Comment by aelred** [comment]
thanks @alexandraorth, now it's your turn @fppt !

**Comment by fppt** [comment]
Something is missing from my components? Will have a look but I have a suspicion a rebase will solve that.

**Comment by fppt** [comment]
Ahhh orient specific classes. Right fixing now.

**Comment by aelred** [comment]
Thanks @fppt . We're done!

File: grakn-factory/orientdb-factory/src/main/java/ai/grakn/factory/OrientDBInternalFactory.java [hunk 1]
```diff
@@ -39,6 +39,18 @@
 
 import static ai.grakn.util.ErrorMessage.INVALID_DATATYPE;
 
+/**
```
Comment (fppt) [inline grakn-factory/orientdb-factory/src/main/java/ai/grakn/factory/OrientDBInternalFactory.java]: One style of class header

File: grakn-graql/src/main/java/ai/grakn/graql/Graql.java [hunk 2]
```diff
@@ -39,6 +39,11 @@
 import java.util.Set;
 import java.util.stream.Stream;
 
+/**
+ * Main class containing static methods for creating Graql queries.
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/Graql.java]: Another style of class header. Note missing author.

File: grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/ClusterMemberMapReduce.java [hunk 3]
```diff
@@ -31,6 +31,17 @@
 
 import static ai.grakn.graql.internal.analytics.Utility.reduceSet;
 
+/**
+ * The MapReduce program for collecting the result of a clustering query.
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/ClusterMemberMapReduce.java]: Another style of class header.

File: grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/fragment/Fragments.java [hunk 4]
```diff
@@ -31,6 +31,9 @@
 
 import static ai.grakn.util.Schema.EdgeLabel.SUB;
 
+/**
+ * Factory for creating instances of {@link Fragment}.
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/fragment/Fragments.java]: And another style of class header.

**Comment by aelred** [comment]
@fppt got author tags now! We'll leave the header format for the future.

**Comment by fppt** [comment]
Thanks @aelred!

**Pull request merged**